enhancements/console/customize-catalogs.md

```diff
@@ -0,0 +1,265 @@
+also dynamically by installing a console plugin or an operator.
+
+Some customers want to align their development process and 
+have teams that support other development teams with their work.
+For this cluster administrators want to hide developer sub-catalog (types) 
+without disabling a full console plugin, for example by uninstalling an operator.
+By hiding for example "Service Backed operators", "Helm", etc. 
+they can still install them via the CLI, see their resources in the topology, 
+but don't show them in the add from developer catalog flow.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide Developer catalog for all users.
+For example for customers who has their own way to 
+install workloads, services, etc. without the developer catalog.
+
+#### Story 2
+
+Customer wants to display only their own customized and validated 
+templates on the dev console but they can't change the default devfiles currently.
+
+As a cluster admin, I want to hide certain sub-catalogs (types) for all users.
+For example for customers who don't want to use Helm charts, 
+or another sub-catalog without disabling Helm support in the whole developer console.
+
+#### Story 3
+
+Customer wants to display only their own customized and validated 
+templates on the dev console.
+
+As a cluster admin, I want to enable certain sub-catalogs (types) for all users.
+
+### Goals
+
+- Allow cluster admins to hide the developer catalog or certain sub-catalogs (types).
+- Allow cluster admins to enable certain sub-catalogs (types) for users.
+- Provide information to admins about sub-catalogs (types) they can hide from users.
+
+### Non-Goals
+
+- Allow cluster admins to customize categories or hide individual items.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`. 
+We can add 
+`spec.customization.developerCatalog.types.disabled` which lists types to hide and 
+`spec.customization.developerCatalog.types.enabled` which lists types to show.
+
+### Workflow Description
+
+```text
+├── spec
+│   ├── customization
+|       ├── developerCatalog
+|           ├── types
+|               ├── disabled
+|               |   └──
+|               ├── enabled
+|                   └──
+└── ...
+```
+
+Each catalog conforms to the following schema:
+
+// +optional
+- `types: object` , defines the types to enable or disable.
+
+// +optional
+- `disabled: string[]` , defines a list of types to hide. 
+
+// +optional
+- `enabled: string[]` , defines a list of types to enable. 
+
+Type(s) listed in `enabled` will only be displayed to the user and remaining type(s) 
+will be hidden. 
+
+If a type is mentioned in both `disabled` and `enabled` then it will be hidden.
+
+If `enabled` is `null`, then all the sub-catalogs(types) will be hidden.
```

Comment:
We have an open question as a part of the API review if this is true or not, IMO an empty list for enabled means all values are displayed, that's the behaviour I have seen elsewhere